lib/StorageStatistics.php

```diff
@@ -65,18 +69,34 @@ protected function countUserEntries() {
 		return (int) $row['num_entries'];
 	}
 
-	/**
-	 * @param string $tableName
-	 * @return int
-	 */
-	protected function countEntries($tableName) {
-		$query = $this->connection->getQueryBuilder();
-		$query->selectAlias($query->createFunction('COUNT(*)'), 'num_entries')
-			->from($tableName);
-		$result = $query->execute();
-		$row = $result->fetch();
-		$result->closeCursor();
-		return (int) $row['num_entries'];
+	protected function getCountOf(string $table): int {
+		return (int)$this->config->getAppValue('serverinfo', 'cached_count_' . $table, '0');
+	}
+
+	public function updateStorageCounts(): void {
+		$storageCount = 0;
+		$fileCount = 0;
+
+		$fileQuery = $this->connection->getQueryBuilder();
+		$fileQuery->select($fileQuery->createFunction('COUNT(*)'))
+			->from('filecache')
+			->where($fileQuery->expr()->eq('storage', $fileQuery->createParameter('storageId')));
+
+		$storageQuery = $this->connection->getQueryBuilder();
+		$storageQuery->selectAlias('numeric_id', 'id')
+			->from('storages');
+		$storageResult = $storageQuery->execute();
+		while ($storageRow = $storageResult->fetch()) {
+			$storageCount++;
+			$fileQuery->setParameter('storageId', $storageRow['id']);
+			$fileResult = $fileQuery->execute();
+			$fileCount += (int)$fileResult->fetchOne();
```

Comment:
fetchOne is not available for 20 :(

Reply:
so adjust on the backport to fetchColumn()

Reply:
Yup, this here goes against 22. Backports get their care when due.